bigframes/core/blocks.py

```diff
@@ -512,9 +512,17 @@ def to_pandas_batches(self):
         """Download results one message at a time."""
         dtypes = dict(zip(self.index_columns, self.index.dtypes))
         dtypes.update(zip(self.value_columns, self.dtypes))
-        results_iterator, _ = self.session._execute(self.expr, sorted=True)
+        _, query_job = self.session._query_to_destination(
+            self.session._to_sql(self.expr, sorted=True),
+            list(self.index_columns),
+            api_name="cached",
+            do_clustering=False,
+        )
+        results_iterator = query_job.result()
         for arrow_table in results_iterator.to_arrow_iterable(
-            bqstorage_client=self.session.bqstoragereadclient
```

Comment:
@tswast Does this mesh with what you know of the python client? It doesn't appear to be streaming one page at a time when the storage feature is being used. https://github.com/googleapis/python-bigquery/blob/cc7b3995eef4f357f031e06389db07f949987eeb/google/cloud/bigquery/_pandas_helpers.py#L816

Reply:
In other words based on what I'm seeing we should either do this or go back to having the page_size parameter, because the current behavior doesn't really work (isn't fully "batched") for large tables. This way seems perhaps better because, as you mentioned, it's counterintuitive for passing page_size to change behavior so drastically. But adding page_size would also make sense to me since it's something people would probably try if they needed to (for example I knew to try it when I encountered slowness this time, at least.)

Reply:
It should be sending one message at a time back via the stream, but I think you're right that there might not be a mechanism to shut it down if we stop iterating over it.

I'd rather we allow `page_size` as a workaround to users than never use the BQ Storage API at all. Let's go back to the way you had it. :-)